integration/spark/src/main/java/openlineage/spark/agent/lifecycle/plan/DatasetSourceVisitor.java

```diff
@@ -61,12 +61,12 @@ public boolean isDefinedAt(LogicalPlan x) {
   }
 
   @Override
-  public List<LineageEvent.Dataset> apply(LogicalPlan x) {
+  public List<OpenLineage.Dataset> apply(LogicalPlan x) {
     DatasetSource datasetSource =
         findDatasetSource(x)
             .orElseThrow(() -> new RuntimeException("Couldn't find DatasetSource in plan " + x));
     return Collections.singletonList(
-        LineageEvent.Dataset.builder()
+        new OpenLineage.InputDatasetBuilder()
```

Comment:
So is the strategy throughout to just always use `InputDataset`, then convert to the appropriate type later? Can you add a comment explicitly stating that? Add a `TODO` to fix this